sos/policies/arch.py

```diff
@@ -0,0 +1,56 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.policies import PackageManager, LinuxPolicy
+from sos.plugins import ArchPlugin
+from sos.utilities import shell_out
+
+
+class ArchPolicy(LinuxPolicy):
+
+    distro = "Arch Linux"
+    vendor = "Arch Linux"
+    vendor_url = "https://www.archlinux.org/"
+    vendor_text = ""
+    # package_manager = PackageManager(
+    #   "pacman --query | awk 'BEGIN {OFS = \"|\"} {print $1,$2}'")
+    valid_subclasses = [ArchPlugin]
+
+    def __init__(self, sysroot=None):
+        super(ArchPolicy, self).__init__(sysroot=sysroot)
+        self.package_manager = Pacman()
+
+    @classmethod
+    def check(cls):
+        """This method checks to see if we are running on Arch.
+            It returns True or False."""
+        try:
+            with open('/etc/os-release', 'r') as f:
+                return "archlinux" in f.read()
+        except IOError:
+            return False
+
+
+class Pacman(PackageManager):
+
+    def get_pkg_list(self):
```

Comment:
Rather a curious question than a comment to change anything: why `get_pkg_list` from parent cant be used? Because it expects packages as strings in format like:

    avalon-framework|4.3|10.el7

while `pacman --query` returns a list of lines like:

    vlc 2.2.2-3

?

So then replacing ' ' by '|' can help in using the parent class method? (then it is imho worth extending the parent class method to use user-defined separator, like:

    def get_pkg_list(self, sep='|'):
        """Returns a dictionary of packages in the following
        format::

            {'package_name': {'name': 'package_name',
                              'version': 'major.minor.version'}}

        """
        if self.query_command:
            cmd = self.query_command
            pkg_list = shell_out(
                cmd, timeout=0, chroot=self.chroot
            ).splitlines()

            for pkg in pkg_list:
                if sep not in pkg:
                    continue
                elif pkg.count(sep) == 1:
                    name, version = pkg.split(sep)
                    release = None
                elif pkg.count(sep) == 2:
                    name, version, release = pkg.split(sep)
                self.packages[name] = {
                    'name': name,
                    'version': version.split(".")
                }
                release = release if release else None
                self.packages[name]['release'] = release

        return self.packages

Would this work?

Reply:
Yes, that's correct. `pacman --query` output is space-delimited and there is no output formatting switch to change this like some other package managers. And yes, being able to specify the delimiter would work. Is that something you'd like to see as part of this PR?

Reply:
I think extending `get_pkg_list` by separator is better option than copying majority of its code. Any thoughts, @bmr-cymru or @TurboTurtle ?